cell_classification/loss.py

```diff
@@ -0,0 +1,63 @@
+import tensorflow as tf
+from tensorflow.keras import backend as K
+
+
+class Loss():
+    """ Wrapper for loss functions that allows for selective masking of the loss
+    """
+    def __init__(self, loss_name, selective_masking, **kwargs):
+        """ Initialize the loss function
+        Args:
+            loss_name (str):
+                name of the loss function
+            selective_masking (bool):
+                whether to use selective masking
+            **kwargs:
+                additional arguments for the loss function
+        """
+        self.loss_fn = getattr(tf.keras.losses, loss_name)(
+            reduction=tf.keras.losses.Reduction.NONE, **kwargs
+        )
+        self.selective_masking = selective_masking
+
+    def mask_out(self, loss_img, y_true):
+        """ Selectively mask the loss by setting it to zero where y_true == -1
+        Args:
+            loss_img (tf.Tensor):
+                loss image
+            y_true (tf.Tensor):
+                ground truth image
+        Returns:
+            tf.Tensor:
+                masked loss image
+        """
+        y_true = tf.reshape(y_true, tf.shape(loss_img))
+        return tf.where(y_true == 2, tf.zeros_like(loss_img), loss_img)
```

Comment:
how come we're checking if `y_true==2`?

Reply:
I replaced the -1 for unspecific markers with 2 in order to store the groundtruth mask in uint8 instead of int32.